File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/buildings/autorisations/AutList.java
package io.github.alexiscomete.lapinousecond.worlds.buildings.autorisations;

import io.github.alexiscomete.lapinousecond.entity.Owner;

import java.util.ArrayList;

public class AutList implements BuildingAutorisation {

    private final ArrayList<BuildingAutorisation> buildingAutorisations = new ArrayList<>();

    @Override
    public boolean isAutorise(Owner owner) {
        for (BuildingAutorisation buildingAutorisation : buildingAutorisations) {
            if (buildingAutorisation.isAutorise(owner)) {
                return true;
            }
        }
        return false;
    }
}

File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/buildings/autorisations/AutList.kt
package io.github.alexiscomete.lapinousecond.worlds.buildings.autorisations

import io.github.alexiscomete.lapinousecond.entity.entities.Owner

open class AutList : BuildingAutorisation {
    protected var buildingAutorisations = ArrayList<BuildingAutorisation>()
    override fun isAutorise(owner: Owner?): Boolean {
        for (buildingAutorisation in buildingAutorisations) {
            if (buildingAutorisation.isAutorise(owner)) {
                return true
            }
        }
        return false
    }
}